qp_klp/AmpliconStep.py

```diff
@@ -0,0 +1,102 @@
+                                       'QCJob',
+                                       project_name,
+                                       'amplicon')
+
+            makedirs(faked_output_folder)
+
+            raw_fastq_files_path = join(self.pipeline.output_path,
+                                        'ConvertJob')
+
+            # get list of all raw output files to be copied.
+            job_output = [join(raw_fastq_files_path, x) for x in
+                          listdir(raw_fastq_files_path)]
+            job_output = [x for x in job_output if isfile(x)]
+            job_output = [x for x in job_output if x.endswith('fastq.gz')]
+            # Undetermined files are very small and should be filtered from
+            # results.
+            job_output = [x for x in job_output if
+                          not basename(x).startswith('Undetermined')]
+
+            # copy the file
+            for fastq_file in job_output:
+                new_path = join(faked_output_folder, basename(fastq_file))
+                copyfile(fastq_file, new_path)
+
+            # FastQC expects the ConvertJob output to also be organized by
+            # project. Since this would entail running the same ConvertJob
+            # multiple times on the same input with just a name-change in
+            # the dummy sample-sheet, we instead perform ConvertJob once
+            # and copy the output from ConvertJob's output folder into
+            # ConvertJob's output folder/project1, project2 ... projectN.
+            faked_output_folder = join(raw_fastq_files_path, project_name)
+            makedirs(faked_output_folder)
+            job_output = [join(raw_fastq_files_path, x) for x in
+                          listdir(raw_fastq_files_path)]
+            job_output = [x for x in job_output if isfile(x)]
+            job_output = [x for x in job_output if x.endswith('fastq.gz')]
+            job_output = [x for x in job_output if
+                          not basename(x).startswith('Undetermined')]
+
+            for raw_fastq_file in job_output:
+                new_path = join(faked_output_folder, basename(raw_fastq_file))
+                copyfile(raw_fastq_file, new_path)
+
+    def generate_reports(self, config, input_file_path):
+        config = self.pipeline.configuration['fastqc']
+        super()._generate_reports(config, self.pipeline.mapping_file)
+
+        return None  # amplicon doesn't need project names
+
+    def generate_prep_file(self):
+        config = self.pipeline.configuration['seqpro']
+
+        seqpro_path = config['seqpro_path'].replace('seqpro', 'seqpro_mf')
+
+        job = super()._generate_prep_file(config,
+                                          self.pipeline.mapping_file,
+                                          seqpro_path,
+                                          self.project_names)
+
+        self.prep_file_paths = job.prep_file_paths
+
+    def generate_commands(self):
```

Comment:
Same here, this should be part of the main class, no?

Reply:
Not sure if you mean the method above generate_commands() or generate_commands() itself, but in both cases each version needs different things done after the base call. In the case of generate_commands(), 'ConvertJob/Logs' needs to be included in the reports-ConvertJob.tgz tarball, whereas that directory is not created by bcl2fastq and therefore the Amplicon version.